### ISSUE TITLE
False positive E308

### ISSUE THREAD

**Issue by fabianvdW** [intellij-rust/intellij-rust#5406]
## Environment

* **IntelliJ Rust plugin version:**: 0.2.122.3105-193
* **Rust toolchain version:**:cargo 1.43.0 (2cbe9048e 2020-05-03), stable-x86_64-pc-windows-gnu (default), rustc 1.43.1 (8d69840ab 2020-05-04)
* **IDE name and version:** IntelliJ IDEA 2019.1.3
* **Operating system:** Windows7

## Problem description
False positive error E308 marked by the IDE

## Steps to reproduce
```rust
pub const fn square(sq: usize) -> u64{
    1u64 << sq
}
```
```
mismatched types [E0308] expected `u64`, found `usize` 
```

**Comment by Undin** [comment]
Try to enable new macro expansion engine. See #3628 for details

**Comment by Undin** [comment]
Duplicate of #5346
